src/full/Agda/TypeChecking/Rules/Term.hs

```diff
@@ -889,6 +892,19 @@ checkExpr e t0 =
                    , nest 2 $ text "cxt =" <+> (prettyTCM =<< getContextTelescope)
                    ]
             coerce v (sort s) t
+
+        A.Generalized s e -> do
+            (t0, t') <- checkGeneralized s $ isType_ e
+            noFunctionsIntoSize t0 t'
```

Comment:
Please do not cut-and paste, rather generalize the code (extract a common auxiliary function).